fakelibusb_test.go

```diff
@@ -109,8 +110,15 @@ func (f *fakeLibusb) setConfig(d *libusbDevHandle, cfg uint8) error {
 	}
 	return nil
 }
-func (f *fakeLibusb) getStringDesc(*libusbDevHandle, int) (string, error) {
-	return "", errors.New("not implemented")
+func (f *fakeLibusb) getStringDesc(d *libusbDevHandle, index int) (string, error) {
+	if dev, ok := f.fakeDevices[f.handles[d]]; ok {
+		if str, ok := dev.strDesc[index]; ok {
```

Comment:
use !ok and unindent the normal case.
i.e.:

str, ok := dev.strDesc[index]
if !ok {
  return "", fmt.Errorf("....")
}
return str, nil

Reply:
done